opus/application/test_api/test_results_api.py

```diff
@@ -174,453 +188,479 @@ def test__api_data_bad_cols4_json(self):
         "[test_results_api.py] /api/metadata_v2: hst-09975-acs-j8n410lbq cats PDS, Hubble Constraints html"
-        url = '/opus/api/metadata_v2/hst-09975-acs-j8n410lbq.html?cats=PDS+Constraints,Hubble+Mission+Constraints'
+        url = '/api/metadata_v2/hst-09975-acs-j8n410lbq.html?cats=PDS+Constraints,Hubble+Mission+Constraints'
         self._run_html_equal_file(url, 'api_metadata2_hst_09975_acs_j8n410lbq_cats_pds_hubble_constraints_html.html')
 
     def test__api_metadata_hst_09975_acs_j8n410lbq_cats_pds_hubble_constraints_csv(self):
         "[test_results_api.py] /api/metadata: hst-09975-acs-j8n410lbq cats PDS, Hubble Constraints csv"
-        url = '/opus/api/metadata/hst-09975-acs-j8n410lbq.csv?cats=PDS+Constraints,Hubble+Mission+Constraints'
+        url = '/api/metadata/hst-09975-acs-j8n410lbq.csv?cats=PDS+Constraints,Hubble+Mission+Constraints'
         self._run_csv_equal_file(url, 'api_metadata_hst_09975_acs_j8n410lbq_cats_pds_hubble_constraints_csv.csv')
 
     def test__api_metadata2_hst_09975_acs_j8n410lbq_cats_pds_hubble_constraints_csv(self):
         "[test_results_api.py] /api/metadata_v2: hst-09975-acs-j8n410lbq cats PDS, Hubble Constraints csv"
-        url = '/opus/api/metadata_v2/hst-09975-acs-j8n410lbq.csv?cats=PDS+Constraints,Hubble+Mission+Constraints'
+        url = '/api/metadata_v2/hst-09975-acs-j8n410lbq.csv?cats=PDS+Constraints,Hubble+Mission+Constraints'
         self._run_csv_equal_file(url, 'api_metadata2_hst_09975_acs_j8n410lbq_cats_pds_hubble_constraints_csv.csv')
 
     # Bad queries
     def test__api_metadata_bad_opusid_json(self):
         "[test_results_api.py] /api/metadata: bad opusid json"
-        url = '/opus/api/metadata/vg-iss-2-s-c4360845x.json'
-        self._run_status_equal(url, 404, settings.HTTP404_UNKNOWN_OPUS_ID)
+        url = '/api/metadata/vg-iss-2-s-c4360845x.json'
+        self._run_status_equal(url, 404,
+                        HTTP404_UNKNOWN_OPUS_ID('vg-iss-2-s-c4360845x',
+                                    '/api/metadata/vg-iss-2-s-c4360845x.json'))
 
     def test__api_metadata_bad_ringobsid_json(self):
         "[test_results_api.py] /api/metadata: bad ringobsid json"
-        url = '/opus/api/metadata/S_IMG_VG2_ISS_4360846_N.json'
-        self._run_status_equal(url, 404, settings.HTTP404_UNKNOWN_RING_OBS_ID)
+        url = '/api/metadata/S_IMG_VG2_ISS_4360846_N.json'
+        self._run_status_equal(url, 404,
+                        HTTP404_UNKNOWN_RING_OBS_ID(None,
+                                '/api/metadata/S_IMG_VG2_ISS_4360846_N.json'))
 
     def test__api_metadata2_bad_opusid_json(self):
         "[test_results_api.py] /api/metadata_v2: bad opusid json"
-        url = '/opus/api/metadata_v2/vg-iss-2-s-,c4360845.json'
-        self._run_status_equal(url, 404, settings.HTTP404_UNKNOWN_OPUS_ID)
+        url = '/api/metadata_v2/vg-iss-2-s-,c4360845.json'
+        self._run_status_equal(url, 404)
```

Comment:
Actually in this case, no. The URL with the comma in it doesn't match the Django regex because we only allow letters, numbers, and dashes in opus ids. So Django itself throws a plain 404 error with no special message.